### DISCUSSION TITLE
Get sunrise and sunset times

### DISCUSSION THREAD

**Discussion by pedrokkrause** [Stellarium/stellarium#2331] [Q&A]
Is there a way to get the sunrise and sunset times throughout the year in the same way that it's possible to get the solar noon times in Astronomical calculations > Transits? I have seen scripts made for Stellarium that do provide these times for a given day, but they don't actually match the times the sun crosses the 0º altitude in Stellarium. I am trying to get this information from Stellarium because it has an accuracy within seconds (not considering the atmosphere), which I need to test an algorithm I developed to calculate sunrise and sunset times. 

An option that I've tried to implement is to use the function getRTSTime(), however, I do not know how to use it in a script. Another option, that I managed to implement, is the following code:
```
var info = core.getObjectInfo("Sun");
var rts = info.rise;
core.debug(rts)
```
But it doesn't return seconds, only HM format

**Comment by alex-w** [comment]
Please check https://github.com/Stellarium/stellarium-data/releases/tag/weekly-snapshot

**Reply by pedrokkrause** [reply]
Thank you, that truly helps a lot! The only thing is that its definition of rise and set is when the first/last edge of the sun crosses the 0º altitude of the observer, which differs a little bit from my definition, which is with respect to the center of it.

**Reply by gzotti** [reply]
Well, the rest of the community defines sunrise as the moment when the upper limb crosses the horizon. This is what can be observed by inhabitants of this planet. 

What you need is the purely geometric semidiurnal arc. We may consider further extensions or configurable options like definition of a sunrise in a later version.

**Reply by pedrokkrause** [reply]
That would be truly useful to have.

**Reply by pedrokkrause** [reply]
Is there a way to change the code such that I could change the rise and set definition?

**Reply by gzotti** [reply]
Short: Yes.
Long: See above: "We may consider...".

**Reply by pedrokkrause** [reply]
What I meant is if there was a way for me to code it myself, but I already managed it, even if not in the most optimal way:
```
core.setDate("2022-01-01T09:00:00")
core.setMinFps(100)
while (1 == 1){
	var info = core.getObjectInfo("Sun");
	var alt = info.altitude;
	if (Math.abs(alt) < 0.0021){
		core.setDate("+1 seconds");
		core.wait(0.01);
		var alt2 = core.getObjectInfo("Sun").altitude;
		if (Math.abs(alt2) < Math.abs(alt)){
			core.debug(core.getDate());
		} else{
			core.setDate("-1 seconds");
			core.debug(core.getDate());
		}
		core.setDate("+23.9833333333333333333333 hours");
	}
	core.setDate("+1 seconds");
	core.wait(0.01);
}
```

**Comment by gzotti** [comment]
Limiting times of sunrise/sunset to minutes is done in an attempt to prevent anybody from believing it can be computed to seconds. The atmosphere is too variable for such hopes.

**Reply by pedrokkrause** [reply]
That may be true if one wants to consider the atmosphere. However, I want to get the sunrise and sunset times _without_ the atmosphere.
